flycheck-rust.el

```diff
@@ -94,6 +95,32 @@ Return non-nil if PROJECT-ROOT is a binary crate, nil otherwise."
   (let ((root-dir (file-name-directory project-root)))
     (file-exists-p (expand-file-name "src/main.rs" root-dir))))
 
+(defun flycheck-rust-find-target (file-name)
+  "Find and return the cargo target associated with the given file.
+
+FILE-NAME is the name of the file that is matched against the
+'src_path' value in the list 'targets' returned by 'cargo
+read-manifest'.  If there is no match, the first target is
+returned by default.
+
+Return a cons cell (TYPE . NAME), where TYPE is the target
+type (lib or bin), and NAME the target name (usually, the crate
+name)."
+  (let* ((manifest (let ((json-array-type 'list))
+                     (json-read-from-string
+                      (shell-command-to-string "cargo read-manifest"))))
```

Comment:
Ok, skipping the shell is indeed best.

Out of curiosity, would `with-output-to-string` work as well, or are there downsides?